### ISSUE TITLE
`mistyped_literal_suffixes` false positive and bad suggestion

### ISSUE THREAD

**Issue by mikerite** [rust-lang/rust-clippy#6129]
```rust
 let _ = 123_32.123;
```

I expected to see this happen: `mistyped_literal_suffixes` not to fire.

Instead, this happened: 
```
  |
8 |     let _ = 123_32.123;
  |             ^^^^^^^^^^ help: did you mean to write: `123.123_f32`
  |
```

**Comment by flip1995** [comment]
I don't get this suggestion on [Playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=038d7b6083bf17c6c51c6602651319d7). Am I missing something?

**Comment by ThibsG** [comment]
Now the suggestion fires on playground.
